src/components/button/button.svelte

```diff
@@ -76,10 +76,26 @@
   class:isTiny={size === 'tiny'}
   class:fab
   class:isLoading
+  disabled={isLoading || isDisabled || undefined}
   on:click={onClick}
   {...$$restProps}
 >
-  <slot>Leo Button</slot>
+  {#if isLoading}
+    {#if $$slots.loading}
+      <slot name="loading" />
+    {:else if !fab}
+      <div>
+        <slot>Leo Button</slot>
+      </div>
+    {/if}
+    <ProgressRing />
+  {:else}
+    <slot name="left-icon" />
+    <div>
+      <slot>Leo Button</slot>
+    </div>
+    <slot name="right-icon" />
```

Comment:
I'd rather call this `icon-before` and `icon-after` to avoid LTR / RTL confusion

Reply:
Won't `-before` and `-after` cause more confusion since the `icon-before` will show up "after" the text in a RTL language?

Reply:
I don't have experience working with RTL, so ultimately, I'm happy to go with whatever.

Reply:
In RTL the before version should still appear _before_ the text, i.e. to the right of the text. `start` and `end` would also be fine.

Reply:
e.g. look what's happening here:
<img width="629" alt="image" src="https://github.com/brave/leo/assets/741836/f012b3b8-a3e3-453a-ae30-52e17264d01b">

Reply:
Thanks!